includes/sanitizers/class-amp-block-sanitizer.php

```diff
@@ -41,27 +41,32 @@ public function sanitize() {
 			$node = $nodes->item( $i );
 
 			// We are only looking for <figure> elements which have wp-block-embed as class.
-			$class = (string) $node->getAttribute( 'class' );
-			if ( false === strpos( $class, 'wp-block-embed' ) ) {
+			$classes = preg_split( '/\s+/', trim( $node->getAttribute( 'class' ) ) );
+
+			if ( ! in_array( 'wp-block-embed', $classes, true ) ) {
 				continue;
 			}
 
-			// Remove classes like wp-embed-aspect-16-9 since responsive layout is handled by AMP's layout system.
+			// Remove classes pertaining to aspect ratios as the embed's responsiveness will be handled by AMP's layout system.
+			if ( in_array( 'wp-has-aspect-ratio', $classes, true ) ) {
+				$classes = array_diff( $classes, [ 'wp-has-aspect-ratio' ] );
+			}
```

Comment:
I think this could be simplified to just:

```suggestion
			$classes = array_diff( $classes, [ 'wp-has-aspect-ratio' ] );
```

Reply:
Actually this could be removed entirely in favor of the use of my proposed `array_filter()` below.